Studies/python/BayesianOptimizationCustom.py

```diff
@@ -18,16 +18,24 @@ def __init__(self, param_ranges_json, params_init_probe_json, fn, probed_points_
         self.num_iter = num_iter
         self.random_state = random_state
         self.params_ranges = {}
+        self.const_params = {}
         self.probed_points = []
         self.probed_points_opt = []
         self.probed_points_target_output =  probed_points_target_output
         self.probed_points_opt_output = probed_points_opt_output
 
         for key,values in self.params_typed_range.items():
-            if values['type'] in ["int", "float"]:
-                self.params_ranges[key] = tuple(values['range'])
-            elif  values['type'] ==  "list" :
-                self.params_ranges[key] = (0, len(values['range']) -1 )
+            if type(self.params_typed_range[key]['range']) == list:
+                if values['type'] in ["int", "float"]:
+                    self.params_ranges[key] = tuple(values['range'])
+                elif  values['type'] ==  "list" :
+                    self.params_ranges[key] = (0, len(values['range']) -1 )
+            else:
+                if values['type'] in ["int", "float"]:
+                    self.const_params[key] = values['range']
+                elif  values['type'] ==  "list" :
+                    self.const_params[key] = values['range']
```

Comment:
`if` and `elif` are not needed here, simply `self.const_params[key] = values['range']`